src/workerd/io/worker-entrypoint.c++

```diff
@@ -165,6 +166,13 @@ kj::Promise<void> WorkerEntrypoint::request(
         cfBlobJson, lock, lock.getExportedHandler(entrypointName, context.getActor()));
   }).then([this](api::DeferredProxy<void> deferredProxy) {
     proxyTask = kj::mv(deferredProxy.proxyTask);
+  }).then([&context](){
+    // Run GC if in test (PredictableMode is only set for tests)
+    if (isPredictableModeForTest()){
+      auto lock = context.getWorker().getIsolate().getApiIsolate().lock();
+      auto isolate = lock->v8Isolate;
+      isolate->RequestGarbageCollectionForTesting(v8::Isolate::GarbageCollectionType::kFullGarbageCollection);
+    }
```

Comment:
Wasn't sure what would be better, attaching to the request promise chain or add to `waitUntilTasks`

Reply:
Yeah I think it would be best to wait until `drain()` has completed, below.

Also we should cover the non-HTTP event types (other methods of `WorkerEntrypiont`) as well.

Reply:
Also, let's throw this into a utility method on `jsg::IsolateBase` so that call sites would only need to do something like 

```
  jsg::IsolateBase::from(lock->v8Isolate).requestGcForTesting();
```

Just helps keep direct use of the v8 apis isolated.

Reply:
Or, alternatively, something like an attachment to `drain()` in the spirit of `kj::defer`... e.g.

```
  incomingRequest->drain().attach(jsg::MaybeTestGcScope(...), kj::mv(incomingRequest));
```

Just brainstorming here, tho. Don't feel strongly about it.

Thanks for doing this!

Reply:
> Also, let's throw this into a utility method on `jsg::IsolateBase` so that call sites would only need to do something like
> 
> ```
>   jsg::IsolateBase::from(lock->v8Isolate).requestGcForTesting();
> ```
> 
> Just helps keep direct use of the v8 apis isolated.

Ended up doing this approach